APOLLO-24180: docs regarding server side processing (#73)
lucidworks/connectors-sdk-resources: README.asciidoc

The data persisted by the connectors framework is distributed across the Fusion cluster. Each node holds its primary partition of the data, as well as backups of other partitions.
If a node goes down during a crawl, the data store remains intact and usable. Connector implementations do not need to be concerned with this layer, because it is all handled by Fusion.

== Server Side Processing

> Important point to be considered on server side when building a connector using a sdk:
> * Connector server does not guarantee the order of processing of items such as Candidate, Document, Delete, etc, which are emitted from the connector plugin, therefore,

Reviewer: ```suggestion
An important point to consider when building a connector is that the server does not guarantee ordering of emitted items such as Candidates, Documents, Deletes, etc., when processing. Therefore,
```

> any logic that assumes the item is processed and proceeds to process the same item immediately based upon previous processing results may produce incorrect results.

Reviewer: ```suggestion
any connector logic that depends on precise ordering of processing (including index-pipeline and Solr commits) may produce incorrect results.
```

> For example, when a document replace is immediately followed by a delete by query, and, delete by query depends upon the result of document replace which was just before issued
> to the connector server, may produce incorrect results, if delete by query is processed before a document replace by the connector server.

Reviewer: ```suggestion
For example, when a document replace is immediately followed by a delete-by-query, and the delete-by-query depends on the document replace to be fully processed and committed. If the document commit has not yet occurred, then the delete-by-query may result in the wrong items being deleted.
```
